motr/m0crate/crate_index.c

```diff
@@ -343,6 +382,70 @@ static bool cr_idx_w_rebalance_ops(struct cr_idx_w *w, enum cr_opcode op);
 static int cr_idx_w_common(struct cr_idx_w *w);
 static int cr_idx_w_warmup(struct cr_idx_w *w);
 
+/** capture final results */
+static void cr_time_capture_results(struct cr_time_measure_ctx *t,
+					struct cr_idx_w *ciw)
+{
+	int i 			  	     = 0;
+	struct cr_idx_ops_result *op_results = ciw->ciw_results.ciwr_ops_result;
+	ciw->ciw_results.ciwr_total_time_s = t->elapsed;
+	ciw->ciw_results.ciwr_total_time_m0 = m0_time_sub(t->ts_next, t->ts);
+	ciw->ciw_results.ciwr_time_per_op_ns =  m0_time_nanoseconds(
+			ciw->ciw_results.ciwr_total_time_m0 / ciw->nr_ops_total);
+
+	/** Calculate Results for each op (PUT, GET, NEXT and DEL) */
+	for (i=0; i < CRATE_OP_NR; i++) {
+		if (op_results[i].cior_op_count) {
+			op_results[i].cior_ops_total_time_s =
+				cr_time_in_seconds(op_results[i].cior_ops_total_time_m0);
+			op_results[i].cior_time_per_op_ns = m0_time_nanoseconds(
+				op_results[i].cior_ops_total_time_m0 /
+				op_results[i].cior_op_count);
+		}
+	}
+
+}
+/** RESULT REPORT */
+static void cr_time_measure_report(struct cr_time_measure_ctx *t,
+					struct cr_idx_w w)
+{
+	int i			     	     = 0;
+	struct cr_idx_ops_result *op_results = w.ciw_results.ciwr_ops_result;
+
+	/** Results in parsable format */
+	fprintf(stdout, "result: total_s, %f, avg_time_per_op_ns, %.1f,"
+		" key_size_bytes, %d, value_size_bytes, %d, ops, %d\n",
+		t->elapsed, w.ciw_results.ciwr_time_per_op_ns,
+		w.wit->key_size, w.wit->value_size, w.nr_ops_total);
+
+	for (i=0; i < CRATE_OP_NR; i++) {
```

Comment:
We usually have space between the assignment operation:
```
for (i = 0; i < CRATE_OP_NR; i++) {
```

Reply:
fixed.